noir-circuits/blob/src/main.nr

```diff
@@ -1,108 +1,20 @@
-// };
-
-//*********************************************************************************
-
 // BIG CONFIG *********************************************************************
 
 mod config;
-mod negative_roots;
 
 use crate::{
     config::{
     BigNum, BLS12_381_Fr_Params, F, FIELDS_PER_BLOB, LOG_FIELDS_PER_BLOB, NOIR_FIELDS_PER_BLOB,
-    FIELDS_CARRYING_AN_EXTRA_BIT_PER_BLOB, D, D_INV, ROOTS
-},
-    negative_roots::NEGATIVE_ROOTS
+    FIELDS_CARRYING_AN_EXTRA_BIT_PER_BLOB
+}
 };
 
 //*********************************************************************************
 
 use std::hash::poseidon2;
 
 global LIMB_MAX = 2.pow_32(120);
-
-unconstrained fn __batch_invert_impl<let N: u32>(mut x: [F; N]) -> [F; N] {
-    let mut accumulator: F = BigNum::one();
-
-    let mut temporaries: [F] = &[];
-    for i in 0..x.len() {
-        temporaries = temporaries.push_back(accumulator);
-        if (x[i].__is_zero() == false) {
-            accumulator = accumulator.__mul(x[i]);
-        }
-    }
-
-    accumulator = accumulator.__invmod();
-    let mut T0: F = BigNum::new();
-    for i in 0..x.len() {
-        let idx = x.len() - 1 - i;
-        if (x[idx].__is_zero() == false) {
-            T0 = accumulator.__mul(temporaries[idx]);
-            accumulator = accumulator.__mul(x[idx]);
-            x[idx] = T0;
-        }
-    }
-    x
-}
-
-// Not used because it resulted in "stack too deep", so it's inlined instead.
-unconstrained fn __compute_fracs(z: F) -> [F; FIELDS_PER_BLOB] {
-    let mut denoms: [F; FIELDS_PER_BLOB] = [BigNum::new(); FIELDS_PER_BLOB];
-    for i in 0..FIELDS_PER_BLOB {
-        denoms[i] = z.__add(NEGATIVE_ROOTS[i]); // (z - ω^i)
-    }
-    let inv_denoms = __batch_invert_impl(denoms); // 1 / (z - ω^i), for all i
-
-    let mut fracs: [F; FIELDS_PER_BLOB] = [BigNum::new(); FIELDS_PER_BLOB];
-    for i in 0..FIELDS_PER_BLOB {
-        let inv_denom = inv_denoms[i]; // 1 / (z - ω^i)
-
-        fracs[i] = ROOTS[i].__mul(inv_denom); // ω^i / (z - ω^i)
-    }
-    fracs
-}
-
-unconstrained fn __field_to_bytes(x: Field) -> [u8; 32] {
-    let x_bytes_slice = x.to_be_bytes(32);
-    let mut x_bytes = [0; 32];
-    for i in 0..32 {
-        x_bytes[i] = x_bytes_slice[i];
-    }
-    x_bytes
-}
-
-unconstrained fn __field_to_bignum(x: Field) -> F {
-    let x_bytes = __field_to_bytes(x);
-    BigNum::from_byte_be(x_bytes)
-}
-
-unconstrained fn __field_to_bignum_limbs(x: Field) -> [Field; 3] {
-    __field_to_bignum(x).limbs
-}
-
-// Only works for bignums with modulus larger than the BN Fr size (which is true
-// for the bls12-381 Fr field).
-fn field_to_bignum(x: Field) -> F {
-    let __x_limbs = __field_to_bignum_limbs(x);
-
-    let mut check = __x_limbs[3 - 1];
-    for i in 1..3 {
-        check *= LIMB_MAX;
-        check += __x_limbs[3 - i - 1];
-    }
-    assert(check == x);
-
-    BigNum { limbs: __x_limbs }
-}
+global TWO_POW_255 = 2.pow_32(255);
 
 // DANGER: this assumes the input bignum is <= the Noir field size.
```

Comment:
`Field` overflows silently

Reply:
Oh yeah this is an artifact of late-night stupidity on my part. Thankfully it's just dangling here and not used anywhere.